lib/statsd/instrument/metric.rb

```diff
@@ -31,6 +31,18 @@
 # @see StatsD::Instrument::Backend A StatsD::Instrument::Backend is used to collect metrics.
 #
 class StatsD::Instrument::Metric
+  if Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.4')
```

Comment:
I'd rather use `Regexp.method_defined?(:match?)` as a condition, because RUBY_VERSION check can be tricky with alternative implementations.

Reply:
Mostly I just wanted to make it clear when this could be removed, but I could do feature detection and us a comment for that.  However, we should probably add any alternative ruby versions we want to support to CI (e.g. truffleruby).  I would actually be more concerned with refinements, which would be more complex to implement, than with `Regexp#match?`, which is easy to implement.

Reply:
Yeah but that's my point. I know that jruby / truffle / etc have `Regexp#match?` however they're likely to return something you don't expect in `RUBY_VERSION`.

So I'm worried you might be using a useless refinement on those.

But yeah 👍 on adding them to CI.